server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/IdentitySetContract.scala

```diff
@@ -0,0 +1,618 @@
+        |				"create": {
+        |					"4f28": {
+        |						"name": "valid send from identity",
+        |						"email": "bob@domain.tld"
+        |					},
+        |					"4f29": {
+        |						"name": "valid send from identity",
+        |						"email": "bob-alias@domain.tld"
+        |					}
+        |				}
+        |			},
+        |			"c1"
+        |		]
+        |	]
+        |}""".stripMargin
+
+    val response: String = `given`
+      .body(request)
+      .header(ACCEPT.toString, ACCEPT_RFC8621_VERSION_HEADER)
+    .when
+      .post
+    .`then`
+      .statusCode(SC_OK)
+      .contentType(JSON)
+      .extract
+      .body
+      .asString
+
+    assertThatJson(response)
+      .inPath("methodResponses[0][1]")
+      .isEqualTo(
+        s""" {
+           |	"accountId": "29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6",
+           |	"newState": "2c9f1b12-b35a-43e6-9af2-0106fb53a943",
+           |	"created": {
+           |		"4f28": {
+           |			"textSignature": "",
+           |			"htmlSignature": ""
+           |		},
+           |		"4f29": {
+           |			"textSignature": "",
+           |			"htmlSignature": ""
+           |		}
+           |	}
+           |}""".stripMargin)
+  }
+
+  @Test
+  def setIdentityShouldFailWhenEmailPropertyIsMissing(): Unit = {
+    val request =
+      s"""{
+         |	"using": ["urn:ietf:params:jmap:core"],
+         |	"methodCalls": [
+         |		[
+         |			"Identity/set",
+         |			{
+         |				"accountId": "29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6",
+         |				"create": {
+         |					"4f29": {
+         |						"name": "Bob",
+         |						"replyTo": [{
+         |							"name": "Alice",
+         |							"email": "alice@domain.tld"
+         |						}],
+         |						"bcc": [{
+         |							"name": "David",
+         |							"email": "david@domain.tld"
+         |						}],
+         |						"textSignature": "Some text signature",
+         |						"htmlSignature": "<p>Some html signature</p>"
+         |					}
+         |				}
+         |			},
+         |			"c1"
+         |		]
+         |	]
+         |}""".stripMargin
+
+    val response =  `given`
+      .header(ACCEPT.toString, ACCEPT_RFC8621_VERSION_HEADER)
+      .body(request)
+    .when
+      .post
+    .`then`
+      .statusCode(SC_OK)
+      .contentType(JSON)
+      .extract
+      .body
+      .asString
+
+    assertThatJson(response)
+      .inPath("methodResponses[0][1]")
+      .isEqualTo(
+        s"""{
+           |	"accountId": "29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6",
+           |	"newState": "2c9f1b12-b35a-43e6-9af2-0106fb53a943",
+           |	"notCreated": [
+           |		[
+           |			"4f29",
+           |			{
+           |				"type": "invalidArguments",
+           |				"description": "Missing '/email' property in Identity object"
+           |			}
```

Comment:
```suggestion
           |			{
           |				"type": "invalidArguments",
           |				"description": "Missing '/email' property in Identity object"
           |                            "properties": "email"
           |			}
```

I bet we could easily have the impacted property specified explicitly with the invalidArgument format....